common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java

```diff
@@ -94,30 +94,30 @@
 
     private static final String TAG = MsalBrokerRequestAdapter.class.getName();
 
-    public static Gson sRequestAdapterGsonInstance;
-
-    static {
-        sRequestAdapterGsonInstance = new GsonBuilder()
-                .registerTypeAdapter(
-                        AbstractAuthenticationScheme.class,
-                        new AuthenticationSchemeTypeAdapter()
-                ).create();
-    }
+    // TODO: provide this from a factory
+    public static Gson sRequestAdapterGsonInstance = new GsonBuilder()
+                      .registerTypeAdapter(
+                            AbstractAuthenticationScheme.class,
+                            new AuthenticationSchemeTypeAdapter()
+                      ).create();
 
     @Override
     public BrokerRequest brokerRequestFromAcquireTokenParameters(@NonNull final InteractiveTokenCommandParameters parameters) {
         Logger.info(TAG, "Constructing result bundle from AcquireTokenOperationParameters.");
 
-        final BrokerRequest brokerRequest = new BrokerRequest.Builder()
+        final String extraQueryStringParameter = parameters.getExtraQueryStringParameters() != null ?
+                QueryParamsAdapter._toJson(parameters.getExtraQueryStringParameters())
+                : null;
+        final String extraOptions = parameters.getExtraOptions() != null ?
+                QueryParamsAdapter._toJson(parameters.getExtraOptions()) : null;
+        final BrokerRequest brokerRequest = BrokerRequest.builder()
                 .authority(parameters.getAuthority().getAuthorityURL().toString())
                 .scope(TextUtils.join(" ", parameters.getScopes()))
                 .redirect(getRedirectUri(parameters))
                 .clientId(parameters.getClientId())
-                .username(parameters.getLoginHint())
-                .extraQueryStringParameter(
-                        parameters.getExtraQueryStringParameters() != null ?
-                                QueryParamsAdapter._toJson(parameters.getExtraQueryStringParameters())
-                                : null)
+                .userName(parameters.getLoginHint())
+                .extraQueryStringParameter(extraQueryStringParameter)
+                .extraOptions(extraOptions)
```

Comment:
Any changes in this class is a protocol breaking change.

Reply:
Which we're going to have with getPrtSsoCookie, right?

Reply:
I'm OK to back this out, by the way.  But if I had the opportunity to clean it up...

Reply:
as in, the older broker probably wouldn't be aware of these 'options'

Reply:
nvm. For some reason, I thought this is BrokerMsalRequestAdapter :/

Reply:
Yeah, I get that.  As long as it drops them nicely on the floor, that's OK.